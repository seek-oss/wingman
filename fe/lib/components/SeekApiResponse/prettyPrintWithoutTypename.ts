/**
 * Pretty prints a JSON value with recursively stripping of the GraphQL
 * `__typename` field.
 */
export const prettyPrintWithoutTypename = (data: unknown) =>
  JSON.stringify(
    data,
    (name, value) => (name === '__typename' ? undefined : value),
    2,
  );
