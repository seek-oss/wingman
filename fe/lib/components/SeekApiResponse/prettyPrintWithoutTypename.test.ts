import { prettyPrintWithoutTypename } from './prettyPrintWithoutTypename';

describe('prettyPrintWithoutTypename', () => {
  it.each`
    description                         | input
    ${'empty object'}                   | ${{}}
    ${'object without __typenames'}     | ${{ id: { value: 123 } }}
    ${'object with nested __typenames'} | ${{ __typename: 'Event', id: { __typename: 'ObjectIdentifier', value: 123 } }}
  `('$description', ({ input }) =>
    expect(prettyPrintWithoutTypename(input)).toMatchSnapshot(),
  );
});
