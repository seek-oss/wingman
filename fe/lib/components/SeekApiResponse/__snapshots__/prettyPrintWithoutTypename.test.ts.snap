// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettyPrintWithoutTypename empty object 1`] = `"{}"`;

exports[`prettyPrintWithoutTypename object with nested __typenames 1`] = `
"{
  "id": {
    "value": 123
  }
}"
`;

exports[`prettyPrintWithoutTypename object without __typenames 1`] = `
"{
  "id": {
    "value": 123
  }
}"
`;
